[models] add a local-apps-page (#1877)
huggingface/hub-docs: docs/hub/local-apps.md

# Use AI Models Locally

You can run AI models from the Hub locally on your machine. This means that you can benefit from these advantages:

- **Privacy**: You won't be sending your data to a remote server.
- **Speed**: Your hardware is the limiting factor, not the server or connection speed.
- **Control**: You can configure models to your liking.
- **Cost**: You can run models locally without paying for an API provider.

## How to Use Local Apps

Local apps are applications that can run Hugging Face models directly on your machine. To get started:

1. **Enable local apps** in your [Local Apps settings](https://huggingface.co/settings/local-apps).

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/settings.png)

> 1. **Choose a supported model** from the Hub by searching for it.

Reviewer: ```suggestion
1. **Choose a supported model** from the Hub by searching for it. You can filter by `app` in the `Other` section of the navigation bar: 
```

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/search_llamacpp.png)

3. **Select the local app** from the "Use this model" dropdown on the model page.

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/button.png)

4. **Copy and run** the provided command in your terminal.

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/command.png)

## Supported Local Apps

> The best way to check if a local app is supported is to go to the Local Apps settings and see if the app is listed. Here is a quick overview of some of the most popular local apps:

Reviewer: Maybe we could add a link for devs that want to add new local apps?

Reviewer 2: don't think it's necessary, since we add them more opportunistically atm.

<Tip>

To use these local apps, copy the snippets from the model card as above.

</Tip>

### Llama.cpp

> Llama.cpp is a high-performance C/C++ library for running LLMs locally with optimized inference across different hardware. If you are running a CPU, this is the best option.

Reviewer: ```suggestion
Llama.cpp is a high-performance C/C++ library for running LLMs locally with optimized inference across lots of different hardware, including CPUs, CUDA and Metal.
```

The CPU comment sounded a bit limiting to me, as if we wouldn't recommend it for GPUs.

**Advantages:**

> - Extremely fast performance for CPU-based models

Reviewer: ```suggestion
- Extremely fast performance for CPU-based models on multiple CPU families
```

- Low resource usage
- Multiple interface options (CLI, server, Python library)

> - Hardware-optimized for CPU and GPU

Reviewer: ```suggestion
- Hardware-optimized for CPUs and GPUs
```

To use Llama.cpp, navigate to the model card and click "Use this model" and copy the command.

```sh
# Load and run the model:
./llama-server -hf unsloth/gpt-oss-20b-GGUF:Q4_K_M
```

### LM Studio

LM Studio is a desktop application that provides an easy way to download, run, and experiment with local LLMs.

**Advantages:**
- Intuitive graphical interface
- Built-in model browser
- Developer tools and APIs
- Free for personal and commercial use

Navigate to the model card and click "Use this model". LM Studio will open and you can start chatting through the interface.

> ### Jan

Reviewer: maybe above LM Studio?

Jan is an open-source ChatGPT alternative that runs entirely offline with a user-friendly interface.

**Advantages:**

> - Complete privacy (all data stays local)

Reviewer: Perhaps redundant after your general intro

- User-friendly GUI
- Chat with documents and files

> - OpenAI-compatible API server

Reviewer: ```suggestion
- OpenAI-compatible API server, so you can run models and use them from other apps
```

- **Privacy**: You won't be sending your data to a remote server.
- **Speed**: Your hardware is the limiting factor, not the server or connection speed.
- **Control**: You can configure models to your liking.
- **Cost**: You can run models locally without paying for an API provider.

## How to Use Local Apps

Local apps are applications that can run Hugging Face models directly on your machine. To get started:

1. **Enable local apps** in your [Local Apps settings](https://huggingface.co/settings/local-apps).

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/settings.png)

1. **Choose a supported model** from the Hub by searching for it. You can filter by `app` in the `Other` section of the navigation bar: 

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/search_llamacpp.png)

3. **Select the local app** from the "Use this model" dropdown on the model page.

![Local Apps](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/local-apps/button.png)

> 4. **Copy and run** the provided command in your terminal.

Reviewer 3: technically in many (most?) cases the link is a deep-link to open the app with the specified model, not a command to copy/paste

(cc @gary149 too for viz)
